### ISSUE TITLE
Prometheus output: Collector as metric name

### ISSUE THREAD

**Issue by pflorczyk** [influxdata/telegraf#4825]
## Feature Request
It would be really useful if prometheus output could be configured to output collector name as metric name.

### Proposal:
Allow configuring collector name as metric name (maybe per collector, in vein of collectors_exclude)
 
### Current behavior:
Metrics are exposed as:
```
# HELP pf_entries Telegraf collected metric
# TYPE pf_entries untyped
pf_entries{host="xxx"} 158150
# HELP pf_inserts Telegraf collected metric
# TYPE pf_inserts untyped
pf_inserts{host="xxx"} 1.140919387e+10
# HELP pf_removals Telegraf collected metric
# TYPE pf_removals untyped
pf_removals{host="xxx"} 1.1409035658e+10
# HELP pf_searches Telegraf collected metric
# TYPE pf_searches untyped
pf_searches{host="xxx"} 1.938925787668e+12
```
### Desired behavior:
```
# HELP pf Telegraf collected metric
# TYPE pf untyped
pf{metric='entries',host="xxx"} 158150
pf{metric='inserts',host="xxx"} 1.140919387e+10
pf{metric='removals',host="xxx"} 1.1409035658e+10
pf{metric='searches',host="xxx"} 1.938925787668e+12
```
### Use case:
Allows simpler and more effective querying in prometheus. It also simplifies creating graphs (label field value as legend):
Before:
```
pf_inserts{host="yyy"}
pf_removals{host="yyy"}
pf_searches{host="yyy"}
```
After:
```
pf{host="yyy",metric!="entries"}
```

In other words try creating cpu graph and track it for all architectures ;)
For metrics exported with node_exporter you can just use:
```
avg without (cpu)(irate(node_cpu{job="node",instance="$instance",mode!="idle"}[5m]))
```
In current telegraf implementation one would have to use ~10 queries.

**Comment by danielnelson** [comment]
I would like to restructure the prometheus input and output mapping, but I don't think the desired behavior above matches best practices established by the prometheus project:

https://prometheus.io/docs/practices/naming/#metric-names
